workspace-manager-clienttests/src/main/java/scripts/testscripts/GetDataReferenceByName.java

```diff
@@ -0,0 +1,34 @@
+package scripts.testscripts;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.client.ApiException;
+import bio.terra.workspace.model.CreateDataReferenceRequestBody;
+import bio.terra.workspace.model.DataReferenceDescription;
+import java.util.List;
+import scripts.utils.ClientTestUtils;
+import scripts.utils.WorkspaceFixtureTestScriptBase;
+
+public class GetDataReferenceByName extends WorkspaceFixtureTestScriptBase {
```

Comment:
Should we remove this class from this PR, given the comment here? https://github.com/DataBiosphere/terra-workspace-manager/pull/183#discussion_r555317348

Separate but related - I'm not quite clear on the intended structure of these test classes. On one hand, it looks like we want to write CRUD-like tests for a resource in one test class (example `CreateGetDeleteWorkspace.java` for Workspaces). On the other hand, `GetDataReferenceByName.java` seems to live by itself here. It looks like that was addressed in [#182](https://github.com/DataBiosphere/terra-workspace-manager/pull/182) with [DataReferenceLifecycle.java](https://github.com/DataBiosphere/terra-workspace-manager/pull/182/files#diff-65909047383aed3785c17cb9d2864086cda6b7ccdd58473140e5dff5497e0490). However, the same [#182](https://github.com/DataBiosphere/terra-workspace-manager/pull/182) also had [EnumerateDataReferences.java](https://github.com/DataBiosphere/terra-workspace-manager/pull/182/files#diff-03c853b3e5fec3083a8555b059e2ceec5c9cf80f0d4bdb399b92a84128e86a19) test class which seems to break out on its own, unlike the other DataReference-related tests.

Reply:
Yep, I need to remove that one.

I think we want:
1. tests that cover each of the basic CRUD endpoints we have (e.g. `CreateGetDeleteWorkspace`, `DataReferenceLifecycle`. These tend to get lumped into a single test since they all exercise the same "create -> read -> delete" combination of endpoints.
2. tests that cover individual endpoints, either more exhaustively or because they aren't included in #1. 

In general I prefer the smaller and more focused tests, but I think individually testing each part of a CRUD api ends up with a lot of duplicate tests.

Reply:
That's fair, IMO.

We should watch out for how this approach plays into other things, though...such as reporting test results. In a scenario where `GET workspace` assertion fails, for example, I wonder how that will propagate into test reports/dashboards for stakeholders to consume. Are we going to see that `doUserJourney` (i.e. the test) failed and then go to the logs to see what assertion is the culprit? This is in contrast to when the CRUD actions are tested separately, thereby making it easy to quickly identify what action (i.e. `GET workspace` in this case) failed.

Hopefully, I'm not overthinking this. Let me know what your thoughts are, either here or Slack/Meet/etc.

Reply:
Agreed, you'll have to dig into the test results a bit to see which action actually failed. At the same time, given that the tests for the actions are so similar (e.g. testing GET and DELETE require a call to create in setup) the tests will likely either pass or fail as a group - even if we separate them into separate tests, if they regularly fail together we'd still need to dig deeper into results to see causes.